src/components/FormFooter.tsx

```diff
@@ -0,0 +1,90 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+enum Filters {
+  Active = 'active',
+  Completed = 'completed',
+  All = 'all',
+}
+
+type Props = {
+  todos: Todo[],
+  filter: string,
+  setFilter: (filter: Filters) => void;
+  onClearCompleted: () => void,
+};
+
+export const FormFooter: React.FC<Props> = ({
+  todos, filter, setFilter, onClearCompleted,
+}) => {
+  const completedTodos = todos.filter(todo => todo.completed);
+  const unCompletedTodos = todos.filter(todo => !todo.completed);
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${unCompletedTodos.length} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link',
+            { selected: filter === 'all' },
```

Comment:
idem for following elements
```suggestion
            { selected: filter === Filters.All },
```